ui/pages/asset/components/asset-page.tsx

```diff
@@ -110,14 +97,32 @@ const AssetPage = ({
     account.methods.includes(EthMethod.SignTransaction) ||
     account.methods.includes(EthMethod.SignUserOperation);
 
-  const { chainId, type, symbol, name, image, balance } = asset;
+  const selectedAccountTokenBalancesAcrossChains: Record<string, any> =
+    useSelector(getSelectedAccountTokenBalancesAcrossChains);
+  const marketDataAcrossChains = useSelector(getTokensMarketDataAcrossChains);
+  const currencyRates = useSelector(getCurrencyRates);
+
+  const { chainId, type, symbol, name, image } = asset;
 
   const address =
     type === AssetType.token
       ? toChecksumHexAddress(asset.address)
       : '0x0000000000000000000000000000000000000000';
 
-  const marketData = allMarketData?.[address];
+  const balance =
+    selectedAccountTokenBalancesAcrossChains[chainId]?.[address] || '0.00000';
```

Comment:
Why `0.00000`?

Reply:
Removed this here: https://github.com/MetaMask/metamask-extension/pull/28193/commits/21066ade6e6f566e6a8363c6012071b0f98b299f

We may need to eventually add some fallback here, but I think a nullish value will be okay for now.